### PR TITLE
frontend: set lower bound on select default index

### PR DESCRIPTION
**Pull request by dschaller**
### Description

Add a lower bound on default option index.

### Testing Performed

manual and unit

### CODE REVIEW

File: frontend/packages/core/src/Input/select.tsx [hunk 1]
```diff
@@ -201,28 +201,33 @@ const Select = ({
   options,
   onChange,
 }: SelectProps) => {
-  if (options?.length === undefined || options?.length === 0) {
-    return null;
-  }
-
-  const defaultIdx = defaultOption < options.length ? defaultOption : 0;
+  const [defaultIdx] = React.useState(
```
Comment (danielhochman) [inline frontend/packages/core/src/Input/select.tsx:204]: why is this a hook?

Comment (dschaller) [reply frontend/packages/core/src/Input/select.tsx:204]: To ensure that the following dependent states are properly updated in order

Comment (danielhochman) [reply frontend/packages/core/src/Input/select.tsx:204]: i don't follow, would like to understand for my own personal benefit. what happens then if this is not a hook vs if it is?

Comment (dschaller) [reply frontend/packages/core/src/Input/select.tsx:204]: Ah nevermind, ignore this :)

**Pull request merged**